packages/options/options.pony

```diff
@@ -101,6 +101,14 @@ class Options is Iterator[(ParsedOption | ParseError | None)]
       _arguments.push(arg.clone())
     end
 
+  new from_array(args: Array[String val] box, fatal: Bool = true) =>
```

Comment:
if this was to be accepted, this should be Array[String val] val

and then you can hold to it.

This does a lot of unnecessary allocating and cloning behind the scenes.

If the incoming array is String val items, there is no need to clone them, that's wasted allocations